### PR TITLE
802 implement availability service

### PR DESCRIPTION
**Pull request by anthonyboffee**
@rndthoughts  I thought I should get this merged to avoid complications with multiple feature branches on the go. The code builds but I will need to implement further validation once I've got the validation project in place.

### CODE REVIEW

File: src/Updatedge.net.Tests/AvailabilityServiceTests.cs [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+        private List<string> _userIdList;
+        
+        [SetUp]
+        public void Setup()
+        {
+            // Put Flurl into test mode
+            _httpTest = new HttpTest();
+
+            // instantiate the Availability service
+            _availabilityService = new AvailabilityService("https://localhost/", "1234567890");
```
Comment (rndthoughts) [inline src/Updatedge.net.Tests/AvailabilityServiceTests.cs]: Can I suggest you look at AutoFixture for some of these dummy values?  It's a nice way of generating random structured data.  Not too important here, but useful to have in your armoury :)

https://github.com/AutoFixture/AutoFixture

File: src/Updatedge.net/Services/V1/AvailabilityService.cs [hunk 2]
```diff
@@ -18,8 +19,15 @@ public AvailabilityService(string baseUrl, string apiKey) : base(baseUrl, apiKey
...
-            try
+            try 
             {
+                // Validate start and end dates
+                if (start > end) throw new ApiWrapperException("Start date cannot be after end date");
+                if (end > start.AddHours(24)) throw new ApiWrapperException("End date must be within 24 hours of Start date (inclusive).");
+
+                // Validate worker Ids
+                if (workerIds.Count() == 0) throw new ApiWrapperException("You must supply at least one worker id.");
+
```
Comment (rndthoughts) [inline src/Updatedge.net/Services/V1/AvailabilityService.cs:30]: Just a thought, can we bung these strings in a resource file?  If it's shared across projects that would also be useful future-wise.

**Review by rndthoughts — APPROVED** [review]
All cool - just one comment about resource files.  Have approve

File: src/Updatedge.net/Services/V1/AvailabilityService.cs (hunk 2 above)
Comment (anthonyboffee) [reply src/Updatedge.net/Services/V1/AvailabilityService.cs:30]: ok no probs. I think be changing this quite a bit once I've got the validation class in place, so I can add a resource file then.

**Pull request merged**